### PR TITLE
week 3: presentation proposal

### PR DESCRIPTION
**Pull request by kthfre**
# Assignment Proposal

## Title

Rainbow deployments and its role in modern CD

## Names and KTH ID

  - Minh Quang Nguyen (mqnguyen@kth.se)
  - Fredrik Gölman (golman@kth.se)

## Deadline
- Week 3

## Category
- Presentation

## Description

A brief introduction of the problem it attempts to solve when deploying new software versions. We would try to briefly cover previous (or "lesser") relevant similar methodologies such as blue/green deployment and discuss some limitations that motivate the extension, which is rainbow deployment. Such limitations may include multiple backend services serving an application or running long-term jobs that may not be interrupted blocking deployment of new versions, both cases advocating the need for more colors.

**Relevance**

We believe it's relevant to DevOps in general and in particular to the subject of week 3 as it deals with continuous deployment. We also believe no one has covered it previously in the course, although there is some coverage of blue/green deployment.

### CODE REVIEW

**Comment by algomaster99** [comment]
Hi! The proposal looks good to me! There seems to be some conflicts. It is probably because you did not sync the branch on your fork. Could you please resolve them? I will merge then.

**Comment by bepp-boop** [comment]
hey I am just a bit concern about the presentation, the reason is a different group also mainly talk about Green/Blue dev, from this I am not sure how would we should focus on it as as the rainbow technique is still in it infancy so very little information about it(i.e no research paper into it as well as very little information) as well as the technique itself is quiet simple so we also talk about Green/Blue to explain the technique and expand it with the rainbow deployment. I just want to make sure that it is still a good idea and how would should we work on it to make it more orginal and not repeated information I suppose

**Comment by algomaster99** [comment]
>  the reason is a different group also mainly talk about Green/Blue dev,

Could you please link that PR?

> as well as the technique itself is quiet simple so we also talk about Green/Blue to explain the technique and expand it with the rainbow deployment.

Well, I don't know about rainbow deployment and I will be happy to know more. 

> I just want to make sure that it is still a good idea and how would should we work on it to make it more orginal and not repeated information I suppose

Link the PR and I will help you scope.

**Comment by bepp-boop** [comment]
https://github.com/KTH/devops-course/pull/2412
This is the gitpull, it is also this week

**Comment by bepp-boop** [comment]
also our demo last week we still haven't receive any feedback for it

**Comment by bepp-boop** [comment]
it was the demo proxmox server with CI one

**Comment by algomaster99** [comment]
> https://github.com/KTH/devops-course/pull/2412

This talks about blue-green deployment. So if your presentation can motivate rainbow deployment, it will be great. Ask ChatGPT for help. It can give good pointers about content and then you could study more about it. Demo is also an option. Is there a tool that facilitates rainbow deployment? 

> it was the demo proxmox server with CI one

Yes, I remember. I think you would have been sent a calendar invite to present again, right?

**Comment by sofiabobadilla** [comment]
> also our demo last week we still haven't receive any feedback for it

Hi @bepp-boop ,
You can find the feedback on the grade in canvas. The feedback was delivered on September 4th on the assignment Demos.
Your collaborator should also have this information on canvas.

**Comment by algomaster99** [comment]
Oops, sorry, @bepp-boop I misunderstood your question. Thanks, @sofiabobadilla for correcting me!

**Comment by bepp-boop** [comment]
oh @sofiabobadilla, I didn't know about that it is in the grade. At least for my grade I haven't able to seen the feedback but @kthfre himself think that the feedback is yet to be given as well.

**Comment by Deee92** [comment]
Hi @bepp-boop, please check the demo assignment on Canvas for your group (mqnguyen-golman), and let us know if you see the comments we added on Sep 4.

Update: we have now made these grades visible, thanks @bepp-boop for letting us know!

**Comment by kthfre** [comment]
> Hi! The proposal looks good to me! There seems to be some conflicts. It is probably because you did not sync the branch on your fork. Could you please resolve them? I will merge then.

Sorry for the delay, should be fixed now.

File: contributions/demo/week2/mqnguyen-golman/README.md [hunk 1]
```diff
@@ -2,7 +2,11 @@
 
 ## Title
 
+<<<<<<< HEAD
+Continuous integretion testing of node and service availability of promox instances
+=======
 Continuous integration testing of node and service availability of promox instances
+>>>>>>> upstream/2024
```
Comment (algomaster99) [inline contributions/demo/week2/mqnguyen-golman/README.md]: You missed resolving the conflict here :)

**Review by algomaster99 — APPROVED** [review]
Thanks!

File: contributions/demo/week2/mqnguyen-golman/README.md (hunk 1 above)
Comment (kthfre) [reply contributions/demo/week2/mqnguyen-golman/README.md]: Should be done now.

**Pull request merged**